### ISSUE TITLE
Inaccurate battery indicator

### ISSUE THREAD

**Issue by mgresko8** [ubports/ubuntu-touch#553]
- Device: BQ Aquaris E4.5
- Channel: Stable
- Build: OTA-3

### Expected behavior
Battery indicator displays actual battery level.

### Actual behavior
Battery indcator show very slow discharging until 50%. Then it drops to zero in several minutes. It is very hard to predict when the phone discharges. It is not a battery issue, I believe. the issue lasts from Ubuntu Touch. They fixed it in some OTA release, but then they screwed it up again by another OTA release.

**Comment by mgresko8** [comment]
Nope. The battery lasts for example for a week. The battery life is OK. Just the indicator shows inaccurate values. It is 100% reproducable.

**Comment by paulmueller** [comment]
I observed the same problem. Here are two screenshots:

![battery1](https://user-images.githubusercontent.com/2853022/51639321-e9b84580-1f60-11e9-8bed-67753c8ee36e.png)
![battery2](https://user-images.githubusercontent.com/2853022/51639322-ea50dc00-1f60-11e9-8eea-c46405895144.png)

The phone does not heat up. I never noticed that this bug has been fixed in a previous OTA (although I might have skipped a few in the beginning).
Someone should add the tag krillin.

**Comment by paulmueller** [comment]
I observed this yesterday:
- the percentage seemed to be accurate down to 33%
- after that, the percentage decreased rapidly without using the phone (two sim cards registered, no wireless network enabled)
- when charging the phone, the percentage appeared to increase linearly, which indicates that the percentage displayed is actually correct.

![battery](https://user-images.githubusercontent.com/2853022/52177164-5df2b480-27bd-11e9-92b8-b281235a8d2a.png)

**Comment by UniversalSuperBox** [comment]
There are some cases where battery indicators are locked to 50% or so due to a bug in the code handling the indicator, however this does not seem to be the case here. Ubuntu Touch takes its battery health information from the Linux kernel, the information is rarely faulty in software. I think what you may be seeing is the battery needing calibration.

All "smart battery" tracking systems will eventually become inaccurate as the battery they track becomes degraded over time. This is something you'll notice most with older devices. It's possible that a battery calibration will resolve this issue. See https://batteryuniversity.com/index.php/learn/article/battery_calibration for more details. In particular,

> To maintain accuracy, a smart battery should periodically be calibrated by running the pack down in the device until “Low Battery” appears and then apply a recharge. The full discharge sets the discharge flag and the full charge establishes the charge flag. A linear line forms between these two anchor points that allow SoC estimation. In time, this line gets blurred again and the battery requires recalibration.

If you're having trouble with an intermittent battery indicator, I'd try this process.

I'll close this issue for now, please let me know if a couple of calibration cycles fix the issue. If not we can reopen for more investigation.

**Comment by mgresko8** [comment]
Hello,

this is not a battery degradation over time. It is since beginning. I calibrate battery all the time. The phone lasts quite long time - several days (maybe 6). But it is completelly unpredictable when it discharges.

Marek

**Comment by mgresko8** [comment]
Just to be honest it was like this from the beginning, some Canonical OTA fixed it and then some other OTA make it not work again.

Please reopen.

Marek

**Comment by UniversalSuperBox** [comment]
Alright
